opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1067,17 +1079,75 @@ namespace Opm {
     BlackoilWellModel<TypeTag>::
     computeWellPotentials(std::vector<double>& well_potentials, Opm::DeferredLogger& deferred_logger)
     {
-        Opm::DeferredLogger local_deferredLogger;
         // number of wells and phases
         const int nw = numWells();
         const int np = numPhases();
         well_potentials.resize(nw * np, 0.0);
 
+
+        const int reportStepIdx = ebosSimulator_.episodeIndex();
+        const double invalid_alq = -1e100;
+        const double invalid_vfp = -2147483647;
+        auto well_state_copy = well_state_;
+        const Wells* local_wells = clone_wells(wells());
+        std::vector<WellInterfacePtr> well_container_copy = createWellContainer(reportStepIdx, local_wells, deferred_logger);
+
+
+        // average B factors are required for the convergence checking of well equations
+        // Note: this must be done on all processes, even those with
+        // no wells needing testing, otherwise we will have locking.
+        std::vector< Scalar > B_avg(numComponents(), Scalar() );
+        computeAverageFormationFactor(B_avg);
```

Comment:
Since `B_avg` is re-calculated here, it might not be necessary to make it a member of the `WellInterface`.  If it is seen as  a simplification of the interfaces of the original master branch, maybe it is better to do it in a separate PR so it is easier to review and evaluate?

Reply:
> if it is seen as a simplification of the interfaces of the original master branch, maybe it is better to do it in a separate PR

Is that really necessary. The effect on the results seems to be below the Jenkins tolerance. Since test that don't involve output of well potentials or usage of well potential does not fail. And among the test that fails the reason is the well potential change. If you have another model you are worried about, please tell me. And I can verify that the B_avg change doesn't hurt.

Reply:
> And I can verify that the B_avg change doesn't hurt.

I understand what you mean. But on the other way, does it do any good or we have a good reason to do it?

It will result in a much smaller pull request and we do not need to evaluate the possible effects from it.

Reply:
In my opinion it simplifies the interface. With the improved well potential calculations the B_avg values are needed deep down in the code. I really want to avoid passing around the B_avg values in a lot of files. With 3 wells models and both .hpp and impl_hpp files + the well interface it is significant work to change this. I think we should allow clean-up of the code while adding features. If not, I am afraid that we will never have time to improve the code.